### DISCUSSION TITLE
To Do Email Notifications

### DISCUSSION THREAD

**Discussion by dylanhru** [stayonhire/feedback#8] [Feedback / Issues]
**Kaylen (ABP):**
"It looks like we can set tasks/reminders. Can we allocate these to specific people, and have an email notification sent to them?"

**Comment by dylanhru** [comment]
**Dylan (StayOnHire):**
Not currently. I can do this relatively easily, however, the devil is in the detail.

1. When should these emails send? ie. Immediately, in 5 mins, on the hour, (x) hrs before due etc.
2. How often will the send repeat, if ever?
3. Is this an automated action or would the ability to manually send it be more useful? (Manual method is easier than setting rigid automation rules for every scenario)
4. Do these above values need to be customisable (if they are different in priority, become annoying etc.)?

**Comment by dylanhru** [comment]
**Kaylen (ABP):**
1. Emails to be sent daily at 6am.
2. Emails to be repeated daily until actions have been marked as complete.
3. Automated action.
4. Priority of tasks is not so important. Basically, we want everybody to be sent a daily email with a list of tasks they need to complete. If they have no tasks, they should receive an email that says they have no tasks.

**Comment by dylanhru** [comment]
**Kaylen (ABP):** Priority 7/10

**Reply by KaylenS** [reply]
Hi Dylan,
Could we move this up the priority list please? We're quite eager to get this happening.

**Reply by dylanhru** [reply]
I have marked this up as an 8/10 now. I have a few ideas and will send you and the team an email with details in coming days to confirm it's what you need.

**Comment by dylanhru** [comment]
Hi @KaylenS,

It's looking like we're not going to get enough time to get this in today as much as I would have liked to, Tom and I have been working on a knock-on bug from other changes to the Field Actions app for Wade. 

The feature will take us about a day in total to do and I plan to get it done next week's Feedback Friday (hopefully nothing else priority comes up!).

Email reminders received will look like this:
![Screenshot 2023-10-13 at 2 43 35 pm](https://github.com/stayonhire/feedback/assets/34927957/00f6a2e7-0967-4999-9a03-348836202027)

Minimum settings required to build:
- On/off switch with frequency
- Timezone adjustments of email sending
![Screenshot 2023-10-13 at 2 46 14 pm](https://github.com/stayonhire/feedback/assets/34927957/e62fd11a-8477-4111-a2ae-13164aa41d88)

**Comment by dylanhru** [comment]
Hi @KaylenS,

Did you see these appear in your email this morning? While we didn't get time to add configurable settings in, all the emails sent out this morning looked great (for all companies). 

While the priority order is backwards, we will fix this soon in our timeline of upcoming work.

PS. To dos are now also visible on the mobile Field Actions app :) [m.stayonhire.com](url)

![IMG_5959](https://github.com/stayonhire/feedback/assets/34927957/f9ea0704-6e0c-431f-b60b-f9e877dd1411)

![IMG_5960](https://github.com/stayonhire/feedback/assets/34927957/aa7838e5-4680-4635-8ca0-9de5e8015e00)

**Comment by KaylenS** [comment]
Hi Dylan,

Yes the notifications appeared in my email, and appear to be working really well. Thanks so much for this.

Kind regards,

Kaylen Slattery | Operations Support

M| 0447 119 430 E| ***@***.******@***.***> W| www.abpower.com.au<http://www.abpower.com.au>

[Graphical user interface, website  Description automatically generated]
